src/detect-engine-address.c

```diff
@@ -571,7 +572,8 @@ int DetectAddressParseString(DetectAddress *dd, const char *str)
                         goto error;
                 }
 
-                int cidr = atoi(mask);
+                int cidr;
+                ByteExtractStringInt32(&cidr, 10, 0, (const char *)mask);
```

Comment:
We should check the return value (and in the next one)

Reply:
hmm Jeff suggested that since the very next line of code is checking the range of cidr, we could drop the check. Would you suggest otherwise?

Reply:
> Would you suggest otherwise?

Yes, I suggest :
`if (ByteExtractStringInt32(&cidr, 10, 0, (const char *)mask) < 0 || cidr > 128)`

I think it is more readable to keep using `ByteExtractStringInt32` and such in the same way, ie control their return value.

Furthermore, we will use an uninitialized value if we do not check `ByteExtractStringInt32` return value as it will not always set `cidr`

Reply:
OK. Adding this in the follow up PR